docs/source/conf.py

```diff
@@ -348,10 +348,15 @@ def inject_weight_metadata(app, what, name, obj, options, lines):
             lines.append("")
 
 
-def generate_weights_table(module, table_name, metrics):
+def generate_weights_table(module, table_name, metrics, include_pattern=None, exclude_pattern=None):
     weight_enums = [getattr(module, name) for name in dir(module) if name.endswith("_Weights")]
     weights = [w for weight_enum in weight_enums for w in weight_enum]
 
+    if include_pattern is not None:
+        weights = [w for w in weights if include_pattern in str(w)]
+    if exclude_pattern is not None:
+        weights = [w for w in weights if exclude_pattern not in str(w)]
```

Comment:
Not a die-hard fan of this, but we have to special-case this generation function (or its inputs) in some way. Happy to consider other suggestions.

Reply:
The other alternative makes the code simpler here but moves complexity to the caller, i.e., they could provide a validation pattern as regex.

```python
def generate_weights_table(module, table_name, metrics, include_pattern=".*"):
    weight_enums = [getattr(module, name) for name in dir(module) if name.endswith("_Weights")]
    weights = [w for weight_enum in weight_enums for w in weight_enum if re.match(include_pattern, w)]
    ...
```
regex would include both include and exclude patterns

Reply:
I thought about using regex, but I could not find a `include_pattern` where

```
re.match(include_pattern, w)
```

would be Truthy for weights that *do not* contain `"Keypoint"`. Any pointer @jdsgomes ?

Reply:
I think this should do the trick:

```python
In [59]:  re.match("^((?!Keypoint).)*$", "no_keypoint_in_name")
Out[59]: <re.Match object; span=(0, 19), match='no_keypoint_in_name'>
```

vs 

```python 
In [58]: re.match("^((?!Keypoint).)*$", "xxxKeypoint")
Out[58]: None
```

Reply:
Thanks @jdsgomes , that seems to work indeed. I'm a bit on the fence with this. In general, I try to avoid regex like the plague. This one typically doesn't read easily to me and would require some extra comment IMHO. 

I'll yield to whichever you prefer. Do you have a preference here?

Reply:
since we are including/excluding simple patterns I would agree that regex here might just complicate things and just save a few lines of code, so I would say to leave as is.